blocks/algorithms/__init__.py

```diff
@@ -222,6 +224,29 @@ def compute_step(self, param, grad_wr_param):
         """
         raise NotImplementedError
 
+    def compute_steps(self, gradients):
+        """Build a Theano expression for steps for all parameters.
+
+        Override this method if you want to process the gradient
+        with respect to all parameters as a whole, not parameter-wise.
+
+        Parameters
+        ----------
+        gradients : dict of (:class:`~tensor.TensorSharedVariable`
```

Comment:
I'm a bit confused as to why this would be an "unnecessary complication". I believe that @dwf is simply suggesting you replace this line by `OrderedDict((param, self.compute_step(param, gradients[param]) for param in gradients)`, a very minor change that will make sure if any future refactor that potentially uses this return value directly (instead of looping over it using keys from an ordered list) will still be deterministic.